underscore.js

```diff
@@ -1344,6 +1344,10 @@
     return obj != null && hasOwnProperty.call(obj, key);
   };
 
+  _.isSafeInteger = function(obj) {
+    return _.isNumber(obj) && _.isFinite(obj) && toString.call(parseInt(obj, 10)) === '[object Number]' && Math.floor(obj) === obj;
```

Comment:
Is the third condition really necessary here? It seems kind of redundant with the first. Also, maybe the first is a bit too lenient. The standard seems to suggest that object-wrapped numbers are not allowed, so it should probably be `typeof obj === 'number'`.